test/base-tlsstream-fixture.hpp

```diff
@@ -0,0 +1,110 @@
+/* Icinga 2 | (c) 2025 Icinga GmbH | GPLv2+ */
+
+#ifndef TLSSTREAM_FIXTURE_H
+#define TLSSTREAM_FIXTURE_H
+
+#include "base/io-engine.hpp"
+#include "base/tlsstream.hpp"
+#include "test/remote-certificate-fixture.hpp"
+#include <BoostTestTargetConfig.h>
+#include <future>
+
+namespace icinga {
+
+/**
+ * Creates a pair of TLS Streams on a random unused port.
+ */
+struct TlsStreamFixture : CertificateFixture
+{
+	TlsStreamFixture()
+	{
+		using namespace boost::asio::ip;
+		using handshake_type = boost::asio::ssl::stream_base::handshake_type;
+
+		auto serverCert = EnsureCertFor("server");
+		auto clientCert = EnsureCertFor("client");
+
+		auto& io = IoEngine::Get().GetIoContext();
+
+		m_ClientSslContext = SetupSslContext(clientCert.crtFile, clientCert.keyFile, m_CaCrtFile.string(), "",
+			DEFAULT_TLS_CIPHERS, DEFAULT_TLS_PROTOCOLMIN, DebugInfo());
+		client = Shared<AsioTlsStream>::Make(io, *m_ClientSslContext);
+
+		m_ServerSslContext = SetupSslContext(serverCert.crtFile, serverCert.keyFile, m_CaCrtFile.string(), "",
+			DEFAULT_TLS_CIPHERS, DEFAULT_TLS_PROTOCOLMIN, DebugInfo());
+		server = Shared<AsioTlsStream>::Make(io, *m_ServerSslContext);
+
+		std::promise<void> p;
+
+		tcp::acceptor acceptor{io, tcp::endpoint{address_v4::loopback(), 0}};
+		acceptor.listen();
+		acceptor.async_accept(server->lowest_layer(), [&](const boost::system::error_code& ec) {
+			if (ec) {
+				BOOST_TEST_MESSAGE("Server Accept Error: " + ec.message());
+				return;
+			}
+			server->next_layer().async_handshake(handshake_type::server, [&](const boost::system::error_code& ec) {
+				if (ec) {
+					BOOST_TEST_MESSAGE("Server Handshake Error: " + ec.message());
+					p.set_exception(std::make_exception_ptr(boost::system::system_error{ec}));
+					return;
+				}
+
+				p.set_value();
+			});
+		});
```

Comment:
If you check `IsVerifyOK()` on the client side, you probably want to do it here as well given that you supply a client certificate. (And probably `GetPeerCertificate() != nullptr` as well.)

Reply:
Ah right, wasn't sure what `IsVerifyOK()` would do if no client certificate was presented. But apparently it already does the safer thing.